tests/test_construction_missing_annotations.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+from typing import ForwardRef, Optional
+
+import pytest
+
+from pydantic.main import BaseModel
+
+
+def test_missing_annotation_warning_defaults_to_true():
+
+    with pytest.raises(UserWarning, match='is not fully defined'):
+
+        Bar = ForwardRef('Bar')
```

Comment:
Better to use `create_module` and `from __future__ import annotations`, then just have

```py
from __future__ import annotations
from pydantic import BaseModel

class Foobar(BaseModel):
    a: MissingType
```

Reply:
you can add this self-referencing case as a separate test. Best to match against more of the error message to confirm where the error is happening.